### ISSUE TITLE
Error "type 'String' is not a subtype of type 'int' in type cast" when displaying video 

### ISSUE THREAD

**Issue by abbjetmus** [yh-luo/webviewtube#17]
I have a details screen were I displays a video based on a youtube id, but occationally I get this error from the package:

![image](https://github.com/yh-luo/webviewtube/assets/65755086/76a90145-79ae-4209-a0fd-b234863ccc15)

![image](https://github.com/yh-luo/webviewtube/assets/65755086/ac85833d-7284-4ea5-913d-b2d0fd58e5bf)

Resulting with display "Video is not available":

![image](https://github.com/yh-luo/webviewtube/assets/65755086/ba4b9068-405c-48b4-9123-d3b6c5d58a8a)

The issue is not the video because it works sometimes.
Do you know what the issue can be?

**Comment by yh-luo** [comment]
Hi,
from the stack trace, it looks like there were some problems while playing the video.
Can you provide the video id that causes the problems? Thanks!

**Comment by abbjetmus** [comment]
Don't think it has anything to do with the video, here is a link: [](https://www.youtube.com/watch?v=X4TAAsJ4C3Y&ab_channel=Zmartrest)

I think the issue might be having the WebviewtubePlayer in the build method?
I also experience issues when pushing a screen and poping the screen back to this screen where the player is, then it complains that a controller has already been initialized:

![image](https://github.com/yh-luo/webviewtube/assets/65755086/de7fbc2e-d07c-4b4b-b8c1-95c4a2d8eb9e)
 

```dart class ArticleScreen extends ConsumerStatefulWidget {
  const ArticleScreen({super.key, required this.args});
  static const String route = 'ArticleScreen';
  final EducationArgs args;

  @override
  ConsumerState<ArticleScreen> createState() => _ArticleScreenState();
}

class _ArticleScreenState extends ConsumerState<ArticleScreen> {
  GTodoStatus status = GTodoStatus.NOT_DONE;
  String? articleMarkdownContent;
  String? articleVideoId;
  String? title;
  bool hasStartedVideo = false;
  WebviewtubeController controller = WebviewtubeController();

  final options = const WebviewtubeOptions(
    forceHd: true,
    loop: true,
    interfaceLanguage: 'sv-SE',
    showControls: true,
  );

  @override
  void initState() {
    super.initState();
    controller.addListener(_onVideoStateChange);
  }

  @override
  void dispose() {
    controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {

    return Scaffold(
      body: StreamBuilder<OperationResponse<GArticleData, GArticleVars>>(
        stream: responseStream,
        builder: (context, snapshot) {
  
          var article = snapshot.data!.data!.cms_article;
          var articleVideoId = article.videoId;

          return SingleChildScrollView(
            child: Container(
              color: Colors.white,
              child: Column(
                children: [
                  if (articleVideoId == null)
                    SizedBox(
                      height: 200,
                      width: double.infinity,
                      child: LazyImage(
                        url: LazyImage.largeImageUrlTransform(imageUrl),
                      ),
                    )
                  else
                    WebviewtubePlayer(
                      videoId: articleVideoId ?? '',
                      options: options,
                      controller: controller,
                    ),```

**Comment by abbjetmus** [comment]
Any chance to get som help with this @yh-luo? This started to occur after a Flutter version upgrade.

**Comment by yh-luo** [comment]
Hi,
Sorry for the wait. I've pushed a possible fix for the problem of casting errors. Could you try if `main`fix the problems for you?
The root cause is [the random error that the IFrame API throw](https://stackoverflow.com/questions/27573017/failed-to-execute-postmessage-on-domwindow-https-www-youtube-com-http). I've tried to handle it better.

As for the late initialization errors, it's also fixed!

**Comment by yh-luo** [comment]
This should be fixed in 4267aa6cef29a8cb71c7787411eaea03f78acd4e.
